iot/api/CloudIotMqttExample/Program.cs

```diff
@@ -0,0 +1,645 @@
+        public string deviceId { get; set; }
+
+        [Value(1, HelpText = "The project containing the device registry.", Required = true)]
+        public string projectId { get; set; }
+
+        [Value(2, HelpText = "The region (e.g. us-central1) the registry is located in.", Required = true)]
+        public string regionId { get; set; }
+
+        [Value(3, HelpText = "The registry containing the device.", Required = true)]
+        public string registryId { get; set; }
+
+        [Value(4, HelpText = "The command sent to the device", Required = true)]
+        public string command { get; set; }
+    }
+
+    [Verb("startMqtt", HelpText = "Start a new mqtt device.")]
+    class MqttExampleOptions
+    {
+        [Value(0, HelpText = "The project containing device registry.", Required = true)]
+        public string projectId { get; set; }
+
+        [Value(1, HelpText = "The region (e.g. us-central1) the registry is located in.", Required = true)]
+        public string regionId { get; set; }
+
+        [Value(2, HelpText = "The ID of the registry to create.", Required = true)]
+        public string registryId { get; set; }
+
+        [Value(3, HelpText = "Cloud IoT Core device id.", Required = true)]
+        public string deviceId { get; set; }
+
+        [Value(4, HelpText = "Path to private key file.", Required = true)]
+        public string private_key_file { get; set; }
+
+        [Value(5, HelpText = "Encryption algorithm to use to generate the JWT. Either 'RS256' or 'ES256'.", Required = true)]
+        public string algorithm { get; set; }
+
+        [Value(6, HelpText = "CA root from https://pki.google.com/roots.pem.", Required = true)]
+        public string caCert { get; set; }
+
+        [Value(7, HelpText = "Number of messages to publish.")]
+        public int numMessages { get; set; }
+
+        [Value(8, HelpText = "Indicates whether the message to be published is a telemetry event or a device state message.")]
+        public string messageType { get; set; }
+
+        [Value(9, HelpText = "MQTT bridge hostname.", Default = "mqtt.googleapis.com")]
+        public string mqttBridgeHostname { get; set; }
+
+        [Value(10, HelpText = "MQTT bridge port.", Default = 8883)]
+        public int mqttBridgePort { get; set; }
+
+        [Value(11, HelpText = "Expiration time, in minutes, for JWT tokens.")]
+        public int jwtExpiresMinutes { get; set; }
+
+        [Option(HelpText = "Wait time (in seconds) for commands.", Default = 120)]
+        public int waitTime { get; set; }
+    }
+
+    public class CloudIotMqttExample
+    {
+        /// <summary>
+        /// Creates an authorized Cloud IoT Core Service client using Application
+        /// Default Credentials.
+        /// </summary>
+        /// <returns>an authorized Cloud IoT Core Service client.</returns>
+        // [START iot_create_auth_client]
+        public static CloudIotService CreateAuthorizedClient()
```

Comment:
It's important that every page in the docs that shows a call to `CreateAuthorizedClient()` also contain a definition of `CreateAuthorizedClient()`.  One way to do that is to surround this function definition with multiple, repeated doc tags.  For example, surround the function with // [START iot_create_rsa_device] and // [END iot_create_rsa_device], and it will also show up in the create rsa device sample when it's rendered on the page.

Another way to do it is to have a separate include for iot_create_auth_client on every page.  It's up to you.  But if `CreateAuthorizedClient()` is not rendered on the page, readers will have a hard time finding it.  I've read a lot of user feedback bugs where users couldn't find a function that would have been revealed by simply clicking "View on Github."

Reply:
I'm sorry I don't fully understand what you are suggesting.